core/rawdb/accessors_chain.go

```diff
@@ -1184,33 +1184,28 @@ func WritePendingEpoch(tx kv.RwTx, blockNum uint64, blockHash common.Hash, trans
 	return tx.Put(kv.PendingEpoch, k, transitionProof)
 }
 
-// Transitioned returns true if the block number comes after POS transition
-func Transitioned(db kv.Getter, blockNum uint64) (trans bool, err error) {
-	data, err := db.GetOne(kv.TransitionBlockKey, []byte(kv.TransitionBlockKey))
-	if err != nil {
-		return false, fmt.Errorf("failed ReadTd: %w", err)
-	}
-	if len(data) == 0 {
+// Transitioned returns true if the block number comes after POS transition or is the last POW block
+func Transitioned(db kv.Getter, blockNum uint64, terminalTotalDifficulty *big.Int) (trans bool, err error) {
+	if terminalTotalDifficulty == nil {
 		return false, nil
 	}
-	return blockNum > binary.BigEndian.Uint64(data), nil
-}
 
-// MarkTreansition sets transition to proof-of-stake from the block number
-func MarkTransition(db kv.StatelessRwTx, blockNum uint64) error {
-	data := make([]byte, 8)
-	binary.BigEndian.PutUint64(data, blockNum)
-	// If we already transitioned then we do not update the transition
-	marked, err := db.Has(kv.TransitionBlockKey, []byte(kv.TransitionBlockKey))
-	if err != nil {
-		return err
+	if terminalTotalDifficulty.Cmp(common.Big0) == 0 {
```

Comment:
I think that we should remove this early return. In other words, `Transitioned` should return `false` for bogus headers with 0 difficulty prior to the Merge. Then such headers will be rejected by the validation.